### ISSUE TITLE
Building master results in link errors on g++ 4.8

### ISSUE THREAD

**Issue by tkoolen** [RobotLocomotion/drake#1649]
On 4.8.5:
```
Linking CXX executable ../../bin/testVector
CMakeFiles/testVector.dir/testVector.cpp.o: In function `Drake::getDrakePath()':
testVector.cpp:(.text+0x6b): undefined reference to `__cxa_guard_acquire'
testVector.cpp:(.text+0x90): undefined reference to `std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&)'
testVector.cpp:(.text+0x9a): undefined reference to `__cxa_guard_release'
testVector.cpp:(.text+0xa9): undefined reference to `std::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string()'
testVector.cpp:(.text+0xc6): undefined reference to `__cxa_guard_abort'
```
4.7 and 4.9 are fine.

**Comment by rdeits** [comment]
g++ 4.8 is also broken w.r.t. std::regex, see: https://github.com/RobotLocomotion/drake/issues/1650#issuecomment-175198691

**Comment by RussTedrake** [comment]
now that #1663 is resolved, let's update the minimum gcc version to 4.9.

**Comment by jamiesnape** [comment]
@RussTedrake Would you like the Jenkins builds to be updated to use GCC 4.9?

**Comment by RussTedrake** [comment]
what version is jenkins using?  do we understand why it is building successfully with a more minimal version, while others have had trouble?  I would much prefer to not increment the minimal version if we can avoid it.

**Comment by jamiesnape** [comment]
GCC 4.8 or Clang 3.7. Just responding to your comment above about updating the minimum version. My personal recommendation would be to support GCC 4.8 as it is so commonly used.

**Comment by tkoolen** [comment]
The issue occurred while compiling a mex function, which Jenkins doesn't do yet.

**Comment by billhoffman** [comment]
@RussTedrake can this issue be closed now that 4.9 is the minimum compiler?

**Comment by jamiesnape** [comment]
It will close automatically once #1862 merges.
